## <span style='color:Yellow'> Name : </span> `Amartyaraj Kumar`
#  <span style='color:Yellow'> Roll : </span>  `19029`

### This Notebook has got two parts. 
### You may want to work on the Part I first, which is at the bottom of this page. 
### Then come back to the Part II  in the top.

# Part II Problem Exercises

## Exercise 1, with List
Create a list called `years_list`, starting with the year of your birth, and each year thereafter until the year of your fifth birthday. For example, if you were born in 1980, the list would be `years_list = [1980, 1981, 1982, 1983, 1984, 1985]`.   ((We are NOT going to check if your year of birth in this exercise is matching with your true records.))

In which year in `years_list` was your third birthday? Remember, you were 0 years of age for your first year.

In [1]:
years_list = [2000,2001,2002,2003,2004,2005]
third_birthday = years_list[4]
print(third_birthday)

2004


### Notes on Lists

A list is a sequence of elements ordered by their position. Duplicates are allowed. Lists provide high-performance indexed access. Since the list is a sequential data type, it possesses all the previously described operators for sequences. In addition, the following indexed, 0-based accesses and operations can be performed on lists:

`pop(i)` returns the ith element of the list and removes it from the list. By default (i. e., without specifying the index), the first element is returned.

`list[i] = element` replaces the element at position i with the passed element.

`append(element)` appends the element to the end of the list.

`insert(i, element)` inserts the element at index i in the list.

`extend(other_list)` appends the elements of other_list to the end of the list.

`count(value)` counts how many times the value value occurs in the list.

`index(value)` returns the index where the value value first occurs in the list.

`remove(value)` removes the element with the value value from the list. Only the first one will be deleted if multiple elements with this value exist. If there is no such value, a ValueError is raised.

`reverse()` reverses the order of the elements in the list.

`sort()` sorts the list in ascending order. An inverse (descending) sort is obtained with sort(reverse=True).

A list is a sequence of elements ordered by their position. Duplicates are allowed. Lists provide high-performance indexed access. Since the list is a sequential data type, it possesses all theoperators for sequences. In addition, 0-based accesses and operations can be performed on lists.


## Exercise 2, with List methods
Now create two lists, then add few elements, remove some elements in one of the lists, then `append` one list with another list, `reverse` the list and demonstrate the use of these methods in the lists.
For example,

`numbers = [1, 2, 3, 4]`

`names = ["Peter", "Tim", "Mike", "Tom", "Mike"]`

`names.append("Tom")`

Create such lists, experiment with these methods and print them.

In [1]:
list_A = [1, 2, 5, 9, 17, 'AIIMS', 'IISER', 'IIT']
list_B = [4, 8, 12, 16, 'BITS', 'IIIT', 'CMI', 'ISI']
list_A.append('HCU')
list_B.remove('CMI')
list_A.extend(list_B)
print(list_A)
list_A.reverse()
print(list_A)

[1, 2, 5, 9, 17, 'AIIMS', 'IISER', 'IIT', 'HCU', 4, 8, 12, 16, 'BITS', 'IIIT', 'ISI']
['ISI', 'IIIT', 'BITS', 16, 12, 8, 4, 'HCU', 'IIT', 'IISER', 'AIIMS', 17, 9, 5, 2, 1]


## Exercise 3, with Sets
Find the common elements of two lists, A and B, and return them as a set using 'intersection' method

Examples

list1 = [1, 2, 4, 7, 8] 

list2 = [2, 3, 7, 9] 

result = {2, 7}


In [11]:
list1 = {1,3,5,7,9,11,13,15,17}
list2 = {2,3,4,5,6,7,8,9,10,11,12,13,14,15}
print(list1.intersection(list2))

{3, 5, 7, 9, 11, 13, 15}


## Exercise 4: Create Dictionary
Define a dict, `menu`, that contains the items and prices in rupees on a restaurant menu

- sandwich is 45
- coffee is 22
- juice is 50
- cake is 63

In [11]:
menu = {'Sandwich' : 45, 'Coffee' : 22, 'Juice' : 50, 'Cake' : 63}
print(menu)



{'Sandwich': 45, 'Coffee': 22, 'Juice': 50, 'Cake': 63}


In [17]:
for item in menu.keys():
    print(item, ":", menu[item])

Sandwich : 45
Coffee : 22
Juice : 50
Cake : 63


## Exercise 5: Odds and evens

1. Define two empty lists, `odds` and `evens`.
2. Ask the user, repeatedly, to enter a number.
     - If they enter an empty string, exit the loop
     - If they enter a non-numeric string, scold them and try again
     - Turn the user's input into an integer, and check if it's odd or even. If it's odd, add it to `odds`. If it's even, add it to `evens`.
3. After the loop, print both `odds` and `evens`.  

In [28]:
odd,even = [],[]
user_input = input("enter a number \n")

while user_input != "":
    try:
        user_input = int(user_input)
        if user_input%2 == 0:
            even.append(user_input)
        else:
            odd.append(user_input)
    except ValueError:
        print("Scold")
    user_input = input("enter")

print(odd,even)
    

[123, 123, 15, 16785] [12, 124, 1234]


## Exercise 6: Vowels, digits, and others

1. Define a `counts` dict with three keys: `vowels`, `digits`, and `others`.  All three should have the value 0.
2. Ask the user, repeatedly, to enter a string.
    - If they enter an empty string, stop asking, and print `counts`.
3. Go through each character in the entered string, and check:
    - Is it a vowel? If so, add 1 to `counts['vowels']`
    - Is it a digit? If so, add 1 to `counts['digits']`
    - If neither, add 1 to `counts['others']`.
    
Example:

    Enter string: hello
    Enter string: bye 123
    Enter string: [ENTER]
    {'vowels':3, 'digits':3, 'others':6}

In [24]:
vowels = 0
others = 0
digitS = 0
user_in = input('Type your word here \n' )
word = user_in.strip()

v = list('aeiouAEIOU')
n = list('1234567890')

for alphabet in word:
        if alphabet in v:
            vowels += 1
        elif alphabet in n:
            digitS += 1
        else:
            others += 1
            

print ('No of vowels :', vowels)
print ('No of others :', others)
print ('No of Digits :', digitS)




Type your word here 
Amartyaraj Kumar
No of vowels : 6
No of others : 10
No of Digits : 0


# Part I Practice Exercises

### To make things easier, first work on the Practice Exercises given in the pdf file and enter them in cells below.